step_command.go

```diff
@@ -26,6 +26,12 @@ type Signature struct {
 //
 // Standard caveats apply - see the package comment.
 type CommandStep struct {
+	// Fields common to various step types
+	Key       string   `yaml:"key,omitempty" aliases:"id,identifier"`
+	Name      string   `yaml:"name,omitempty" aliases:"label"`
+	DependsOn []string `yaml:"depends_on,omitempty"`
```

Comment:
sigh...

`depends_on` is not so simple, unfortunately, much though i wish it was. `depends_on` can be any of
- a string of a step key, eg `test` - see screenshot above!
- a sequence of string step keys, like you have here
- for some godforsaken reason, a sequence of the form:
```YAML
depends_on:
  - { step: "some_key" }
  - { step: "some_other_key" }
```

see [Defining Explicit Step Dependencies](https://buildkite.com/docs/pipelines/dependencies#defining-explicit-dependencies) in the docs for examples of this.

Reply:
I'll cut `DependsOn` for now.